webrtc.html

```diff
@@ -2890,6 +2890,79 @@
+          <dd>
+            <p>Attempts to replace the track being sent with another track
+            provided, without renegotiation.</p>
+
+            <p>When the <dfn id="dom-rtpsender-replacetrack"><code title=
+            "">replaceTrack()</code></dfn> method is invoked, the user agent MUST
+            run the following steps:</p>
+
+            <ol>
+              <li>
+                <p>Let <var>p</var> be a new promise.</p>
+              </li>
+
+              <li>
+                <p>Let <var>withTrack</var> be the argument to this method.</p>
+              </li>
+
+              <li>
+                <p>If <var>withTrack's</var> <code><a href=
+                  "getusermedia.html#dom-mediastreamtrack-kind">kind</a></code>
+                attribute differs from the </var><code><a href=
+                  "getusermedia.html#dom-mediastreamtrack-kind">kind</a></code>
+                of this sender's
+                <code><a href="#dom-rtpsender-track">track</a></code>,
+                then reject
+                <var>p</var> with <code>IncompatibleMediaStreamTrackError</code>,
+                return <var>p</var> and abort these steps.</p>
+              </li>
+
+              <li>
+                <p>Run the following steps in parallel:</p>
+              </li>
+              <ol>
+                <li>
+                  <p>Determine if negotiation is needed to transmit
+                  <var>withTrack</var> in place of the sender's existing track.
+                  Ignore which <code>MediaStream</code> the track resides in and
+                  the track's id in this determination. If negotiation is
+                  needed, then reject <var>p</var> with
+                  <code>NegotiationNeededError</code>, return
+                  <var>p</var> and abort these steps.</p>
+                </li>
+                <li>
+                  <p>Have the sender switch seamlessly to transmitting
+                  <var>withTrack</var> in place of what it is sending, without
+                  negotiating. On the remote receiving
+                  end, the track MUST maintain its existing id and stream
+                  associations until the connection ends. To accomplish this,
+                  the sender MUST keep track of the original id and stream
+                  associations, and keep using these in future renegotiations.
+                  </p>
+                </li>
+                <li>
+                  Set this sender's
+                  <code><a href="#dom-rtpsender-track">track</a></code>
+                  attribute to <var>withTrack</var>, and resolve <var>p</var>
+                  with <var>undefined</var>.
+                </li>
+
+              </ol>
+              <li>
+                <p>Return <var>p</var>.</p>
+              </li>
+            </ol>
+
+            <div class="note">
+              <p>Things like dimensions and frame rate do not require negotiation.
```

Comment:
s/do not/might not/  Changing frame rates to values outside of the negotiated imageattr bounds will require negotiation; changing frame rate might cause the block rate for the codec to be exceeded.  Sources that also encode (typically hardware encoders) might be unable to produce the negotiated codec; similarly, software sources might not implement the codec that was negotiated for an encoding source.

Please take another shot at this paragraph.